src/StoreApi/Utilities/ProductQueryFilters.php

```diff
@@ -211,4 +211,54 @@ public function get_rating_counts( $request ) {
 
 		return array_map( 'absint', wp_list_pluck( $results, 'product_count', 'rounded_average_rating' ) );
 	}
+
+	/**
+	 * Gets product by metas.
+	 *
+	 * @since TBD
+	 * @param array $metas Array of metas to query.
+	 * @return array $results
+	 */
+	public function get_product_by_metas( $metas = array() ) {
+		global $wpdb;
+
+		if ( empty( $metas ) ) {
+			return array();
+		}
+
+		$where   = array();
+		$results = array();
+
+		foreach ( $metas as $column => $value ) {
+			if ( 'min_price' === $column ) {
+				$value   = number_format( (float) $value, 4 );
+				$where[] = $wpdb->prepare( "{$column} >= %s", $value ); // phpcs:ignore WordPress.DB.PreparedSQL.InterpolatedNotPrepared
+				continue;
+			}
+
+			if ( 'max_price' === $column ) {
+				$value   = number_format( (float) $value, 4 );
+				$where[] = $wpdb->prepare( "{$column} <= %s", $value ); // phpcs:ignore WordPress.DB.PreparedSQL.InterpolatedNotPrepared
+				continue;
+			}
+
+			$where[] = $wpdb->prepare( "{$column} = %s", $value ); // phpcs:ignore WordPress.DB.PreparedSQL.InterpolatedNotPrepared
+		}
+
+		if ( ! empty( $where ) ) {
+			$where_clause = implode( ' AND ', $where );
+
+			// phpcs:disable
+			$results = $wpdb->get_col(
+				"
+				SELECT DISTINCT `product_id`
+				FROM {$wpdb->prefix}wc_product_meta_lookup
+				WHERE {$where_clause}
+				"
+			);
+			// phpcs:enable
+		}
```

Comment:
What do you think about structuring the query as:

```suggestion
		$params = array();
		foreach ( $metas as $column => $value ) {
			if ( $column === 'min_price' ) {
				$where[]  = "{$column} >= %f";
				$params[] = (float) $value;
				continue;
			}

			if ( $column === 'max_price' ) {
				$where[]  = "{$column} <= %f";
				$params[] = (float) $value;
				continue;
			}

			$where[]  = "{$column} = %s";
			$params[] = $value;
		}

		if ( ! empty( $where ) ) {
			$where_clause = implode( ' AND ', $where );
			// Use a parameterized query
			$results = $wpdb->get_col(
					$wpdb->prepare( "SELECT DISTINCT product_id FROM {$wpdb->prefix}wc_product_meta_lookup WHERE {$where_clause}",
					$params
				)
			);
		}
```
This way we can run `$wpdb->prepare` just once (with a parametized query) instead of multiple times for each individual parts that constitute the `where` clause. 

This also improves security, as with this change, we are now preparing the statement (via `$wpdb->prepare`) for the entire query and not just for the `$where_clause` parts.

Reply:
Yup no objections from me.